From 7c203d8535574623413e54adc41f99bd80dc943b Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Sat, 13 Oct 2012 16:12:22 +0300
Subject: [PATCH 1/9] omap2-clk: Add missing lcdc clock definition

Looks like the lcdc clock definition got dropped.
It is required for the LCD controller to work. Reintroduce.
---
 arch/arm/mach-omap2/clock33xx_data.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/clock33xx_data.c b/arch/arm/mach-omap2/clock33xx_data.c
index 17e3de5..9892ef2 100644
--- a/arch/arm/mach-omap2/clock33xx_data.c
+++ b/arch/arm/mach-omap2/clock33xx_data.c
@@ -866,6 +866,16 @@ static struct clk lcd_gclk = {
 	.recalc		= &followparent_recalc,
 };
 
+static struct clk lcdc_fck = {
+	.name           = "lcdc_fck",
+	.clkdm_name     = "lcdc_clkdm",
+	.parent         = &lcd_gclk,
+	.enable_reg     = AM33XX_CM_PER_LCDC_CLKCTRL,
+	.enable_bit     = AM33XX_MODULEMODE_SWCTRL,
+	.ops            = &clkops_omap2_dflt,
+	.recalc         = &followparent_recalc,
+};
+
 static struct clk mmc_clk = {
 	.name		= "mmc_clk",
 	.clkdm_name	= "l4ls_clkdm",
@@ -1074,6 +1084,7 @@ static struct omap_clk am33xx_clks[] = {
 	CLK(NULL,	"clkout2_ck",		&clkout2_ck,	CK_AM33XX),
 	CLK(NULL,	"timer_32k_ck",		&clkdiv32k_ick,	CK_AM33XX),
 	CLK(NULL,	"timer_sys_ck",		&sys_clkin_ck,	CK_AM33XX),
+	CLK("da8xx_lcdc.0",     NULL,           &lcdc_fck,      CK_AM33XX),
 };
 
 int __init am33xx_clk_init(void)
-- 
1.7.7.6

